### DISCUSSION TITLE
Is there a way to update multiple family providers at once?

### DISCUSSION THREAD

**Discussion by m-haisham** [rrousselGit/riverpod#1427] [Q&A] [answered]
I'm updating a collection of items at once (seen to unseen and vice versa). The item state is being controlled by a `.autoDispose.family` provider keyed by an id. I want to update the state of all of the family providers at once.

As I understand the below code which updates each of the family events as they are called. Which would mean multiple build calls.

```
for (final item in items) {
    _read(itemFamily(item).notifier).setSeen(true)
}
```

**Comment by rrousselGit** [comment, marked as answer]
Your code looks correct

What do you mean by "multiple build calls"?

**Reply by m-haisham** [reply]
It would seem I was mistaken then. I was under the assumption that reassigning the state would trigger rebuild immediately. So the above code would call build only once?

**Reply by rrousselGit** [reply]
Yes.
Your code is basically equivalent to:

```dart
setState(() {});
setState(() {});
```

This rebuilds the widget only once, as both setState invocations happen in the same frame.

If you were to add an async/await between both:

```dart
setState(() {});
await something();
setState(() {});
```

then your widget would rebuild twice. But that's not what you're doing here

**Reply by m-haisham** [reply]
Thank you for clarifying this.
